app/controllers/formbuilder/edit.js

```diff
@@ -157,4 +161,59 @@ export default class FormbuilderEditController extends Controller {
       console.error(`Caught:`, error);
     }
   }
+
+  async getConceptSchemesAsTtlInTtlCode(ttlCode) {
+    const ttlCodeArray = [];
+    const uris = this.getConceptSchemeUrisInTtl(ttlCode);
+
+    if (uris.length == 0) {
+      return;
+    }
+
+    for (const conceptSchemeUri of uris) {
+      const conceptSchemes = await this.store.query('concept-scheme', {
+        include: 'concepts',
+        filter: {
+          ':uri:': conceptSchemeUri,
+        },
+      });
+      const conceptSchemesAsArray = [...conceptSchemes];
+      const ttl = await this.conceptSchemesWithConceptsToTtl(
+        conceptSchemesAsArray
+      );
+      ttlCodeArray.push(ttl);
+    }
+
+    return ttlCodeArray.join('\n');
+  }
+
+  async conceptSchemesWithConceptsToTtl(conceptSchemes) {
+    const ttlArray = [];
+    for (const conceptScheme of conceptSchemes) {
+      ttlArray.push(await conceptScheme.modelWithConceptsAsTtlCode());
+    }
+
+    return ttlArray.join(' ');
+  }
+
+  getConceptSchemeUrisInTtl(store) {
+    const formOptions = store.match(
+      undefined,
+      FORM('options'),
+      undefined,
+      this.model.graphs.formGraph
+    );
+
+    const conceptSchemeUris = [];
+    for (const triple of formOptions) {
+      const optionsAsString = triple.object;
+      try {
+        const jsonOptions = JSON.parse(optionsAsString);
+        conceptSchemeUris.push(jsonOptions.conceptScheme);
+      } catch (error) {
+        return;
+      }
+    }
+    return conceptSchemeUris;
+  }
```

Comment:
Maybe split this up into smaller util methods? The reason for this would be that this is logic for one specifi subject. When calling one or two methods in this controller it will be easier to read and understand what is happening?

Reply:
I dont think that is nescessary, if later on we do want to reuse some smaller piece of code, there we actually turn it unto a util. For now this is good :)